### PR TITLE
if user has dropship access then user can set recuitemnt dropship key

### PR DESCRIPTION
**Pull request by achmadfatoni**
@mrwaim review this one.

### CODE REVIEW

File: routes/routes.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-Route::get('/join/{recruitment_key}', ['as' => 'recruitment.join', 'uses' => '\Klsandbox\RecruitmentRoute\Http\Controllers\RecruitmentManagementController@getJoin']);
+Route::get('/join/{recruitment_key}/{role}', ['as' => 'recruitment.join', 'uses' => '\Klsandbox\RecruitmentRoute\Http\Controllers\RecruitmentManagementController@getJoin']);
```
Comment (mrwaim) [inline routes/routes.php]: Dont do this - it make it look ugly. The role should come from the recruitment key itself

File: views/settings.blade.php [hunk 2]
```diff
@@ -47,7 +58,19 @@
                     </header>
                     <div class="panel-body">
                         <span class="amount reg-list-link"><a id="recruitment_link"
-                                                              href='{{ url('/join/' . $recruitment_key) }}'>{{ url('/join/' . $recruitment_key) }}</a></span>
+                                                              href='{{ url('/join/' . $recruitment_key .'/stockist') }}'>{{ url('/join/' . $recruitment_key .'/stockist') }}</a></span>
+                    </div>
+                @endif
+
+                @if($auth->dropship && $recruitment_dropship_key)
```
Comment (mrwaim) [inline views/settings.blade.php:68]: Server side, do we check the user is able to refer, before allowing them to refer?

**Comment by achmadfatoni** [comment]
@mrwaim pull request updated.

File: src/Klsandbox/RecruitmentRoute/Http/Controllers/RecruitmentManagementController.php [hunk 3]
```diff
@@ -29,11 +30,21 @@ public function __construct()
...
+        $id = $user->id;
+
+        $rules = [
+            'recruitment_key' => 'required|min:5|max:300|alpha_dash|unique:users,recruitment_key,' . $id . ',id|recruitment_stockist',
+
+        ];
+        if ($user->access()->dropship) {
+            $rules = [
+                'recruitment_key' => 'required|min:5|max:300|alpha_dash||different:recruitment_dropship_key|unique:users,recruitment_key,' . $id . ',id|recruitment_stockist:'.$id,
+                'recruitment_dropship_key' => 'required|min:5|max:300|alpha_dash||different:recruitment_key|unique:users,recruitment_dropship_key,' . $id . ',id|recruitment_dropship:'.$id
```
Comment (mrwaim) [inline src/Klsandbox/RecruitmentRoute/Http/Controllers/RecruitmentManagementController.php:43]: ||

File: src/Klsandbox/RecruitmentRoute/Http/Controllers/RecruitmentManagementController.php [hunk 4]
```diff
@@ -151,10 +182,17 @@ public function postPhone(JoinPhonePostRequest $request)
         $user = User::findWithHash($user_hash);
         Site::protect($user);
 
+        $role = Role::findByName(Input::get('role'));
```
Comment (mrwaim) [inline src/Klsandbox/RecruitmentRoute/Http/Controllers/RecruitmentManagementController.php:185]: Duplicate

**Pull request merged**